ice-adapter/src/main/java/com/faforever/iceadapter/IceAdapter.java

```diff
@@ -20,62 +21,41 @@
         description = "An ice (RFC 5245) based network bridge between FAF client and ForgedAlliance.exe")
 @Slf4j
 public class IceAdapter implements Callable<Integer> {
-    @CommandLine.ArgGroup(exclusive = false)
-    private IceOptions iceOptions;
-
-    public static String TELEMETRY_SERVER;
-
-    public static boolean ALLOW_HOST = true;
-    public static boolean ALLOW_REFLEXIVE = true;
-    public static boolean ALLOW_RELAY = true;
-
-    public static volatile boolean running = true;
-
+    private static IceAdapter INSTANCE;
     public static String VERSION = "SNAPSHOT";
+    public static volatile GameSession gameSession;
 
-    public static int id = -1;
-    public static int gameId = -1;
-    public static String login;
-    public static int RPC_PORT;
-    public static int GPGNET_PORT = 0;
-    public static int LOBBY_PORT = 0;
+    @CommandLine.ArgGroup(exclusive = false)
+    private IceOptions iceOptions;
 
-    public static int PING_COUNT = 1;
-    public static double ACCEPTABLE_LATENCY = 250.0;
-
-    public static volatile GameSession gameSession;
+    public volatile boolean running = true;
 
     public static void main(String[] args) {
         new CommandLine(new IceAdapter()).setUnmatchedArgumentsAllowed(true).execute(args);
     }
 
     @Override
     public Integer call() {
-        IceAdapter.start(iceOptions);
+        INSTANCE = this;
+
+        start();
         return 0;
     }
 
-    public static void start(IceOptions iceOptions) {
+    public void start() {
         determineVersion();
+        log.info("Version: {}", VERSION);
 
-        loadOptions(iceOptions);
+        Debug.DELAY_UI_MS = iceOptions.getDelayUi();
+        Debug.ENABLE_DEBUG_WINDOW = iceOptions.isDebugWindow();
+        Debug.ENABLE_INFO_WINDOW = iceOptions.isInfoWindow();
```

Comment:
Why do we create a new static variable if we have it in IceOptions?

Reply:
Will come in a later commit ;) One thing at a time.